demisto_sdk/commands/validate/tests/IF_validators_test.py

```diff
@@ -0,0 +1,332 @@
+    When:
+        - run is_valid method
+    Then:
+        Case 1:
+            - Ensure the error message is as expected
+            - Ensure number of failure is as expected
+        Case 2:
+            - Ensure the error message is as expected with the bad words list
+    """
+    results = IsValidNameAndCliNameValidator().is_valid(content_items=content_items)
+    assert results[0].message == expected_msg
+
+
+def test_IsValidContentFieldValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'content' field is set to False
+    """
+    content_items: List[IncidentField] = [
+        create_incident_field_object(["content"], [False])
+    ]
+
+    results = IsValidContentFieldValidator().is_valid(content_items)
+    assert results[0].message == "The `content` key must be set to true."
+
+
+def test_IsValidSystemFlagValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'system' field is set to True
+    """
+    content_items: List[IncidentField] = [
+        create_incident_field_object(["system"], [True])
+    ]
+
+    results = IsValidSystemFlagValidator().is_valid(content_items)
+    assert results[0].message == "The `system` key must be set to false."
+
+
+def test_IsValidFieldTypeValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'type' field is not valid
+    """
+    content_items: List[IncidentField] = [
+        create_incident_field_object(["type"], ["test"])
+    ]
+
+    results = IsValidFieldTypeValidator().is_valid(content_items)
+    assert (
+        results[0].message
+        == f"Type: `test` is not one of available types.\navailable types: {FIELD_TYPES}."
+    )
+
+
+def test_IsValidGroupFieldValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'group' field is not valid
+    """
+    content_items: List[IncidentField] = [create_incident_field_object(["group"], [2])]
+
+    results = IsValidGroupFieldValidator().is_valid(content_items)
+    assert results[0].message == "Group 2 is not a group field."
+
+
+def test_IsCliNameFieldAlphanumericValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'cliName' value is non-alphanumeric
```

Comment:
What about checking uppercase?
Let's add parameterization of `"_", "foo_", "Foo", "123_", "123A" to this test